Feature/WebEngine support (#187)
smartdevicelink/sdl_hmi_integration_guidelines: docs/BasicCommunication/SetAppProperties/index.md

## SetAppProperties

Type
: Function

Sender
: HMI

Purpose
: Enable/disable an application and set authentication data

!!! must

1. Send `SetAppProperties` (enabled=true) to install a web engine app.

!!!

!!! may
* Send an [UpdateSDL](../../sdl/updatesdl) RPC after setting app properties to force SDL to perform a PTU.
* Send `SetAppProperties` (enabled=false) to add web engine apps to SDL that won't be installed on the HMI

!!!

!!! info

* The OEM store uses app data in SDL Server (synchronized from SHAID) for SetAppProperties.
* If SDL does not know the `appID` it should trigger a PTU.

!!!

### Request

#### Parameters
|Name|Type|Mandatory|Additional|
|:---|:---|:--------|:---------|
|properties|[Common.AppProperties](../../common/structs/#appproperties)|true||

### Response

#### Parameters

This RPC has no additional parameter requirements.

### Example Request
```json
{
  "id":47,
  "jsonrpc": "2.0",
  "method" : "BasicCommunication.SetAppProperties",
  "params" : {

>     "properties": [

Reviewer: Properties here should not be an array, just a single object.

Reviewer 2: hello @iCollin,
please see updates in 69f87d9f974014fc0a11c6185c696554c4becbaa

## SetAppProperties

Type
: Function

Sender
: HMI

Purpose
: Enable/disable an application and set authentication data

!!! must

> 1. Send `SetAppProperties`(enabled=true) if webEngine app is installed.

Reviewer: ```suggestion
1. Send `SetAppProperties` (enabled=true) to install a web engine app.
```
i think something like this is more clear

!!!

!!! may

> * Request `UpdateSDL` after setting app properties to enforce SDL to perform PTU.

Reviewer: ```suggestion
* Send an `[UpdateSDL](../../sd/updatesdl)` RPC after setting app properties to force SDL to perform a PTU.
```
not sure if the link works

Author: please, see updates in https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/187/commits/4ebba070bd78b95c3a00721cc2bcb639a6dfb422

> * Send `SetAppProperties`(enabled=false) to add not-installed webEngine apps to SDL

Reviewer: ```suggestion
* Send `SetAppProperties` (enabled=false) to add web engine apps to SDL that won't be installed on the HMI
```

!!!

!!! info

* The OEM store uses app data in SDL Server (synchronized from SHAID) for SetAppProperties.
* If SDL does not know the `appID` it should trigger a PTU.

!!!

### Request

#### Parameters
|Name|Type|Mandatory|Additional|
|:---|:---|:--------|:---------|
|properties|[Common.AppProperties](../../common/structs/#appproperties)|true||

### Response

#### Parameters

This RPC has no additional parameter requirements.

### Example Request

> ```json

Reviewer: Please clean up the alignment of this JSON, I left some more specific comments on the `GetAppProperties` JSON.

Author: please, see updates in https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/187/commits/4ebba070bd78b95c3a00721cc2bcb639a6dfb422

{
  "id" : 47,
  "jsonrpc" : "2.0",
  "method" : "BasicCommunication.SetAppProperties",
  "params" :
  {
    "properties": {
          "nicknames":
          {
              "Hello Sdl"
      },
      "policyAppID" : 123456, 
      "enabled": true, 
      "transportType":"WS", 
      "hybridAppPreference": "CLOUD"
      },
    "code" : 0,
    "method" : "BasicCommunication.GetAppProperties"
    }
}
```

### Example Response

```json
{
  "id" : 47,
  "jsonrpc" : "2.0",
  "result" : 
  {
    "code" : 0,
    "method" : "BasicCommunication.SetAppProperties",

>     }

Reviewer: please align closing bracket with opening bracket

Author: please, see updates in https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/187/commits/4ebba070bd78b95c3a00721cc2bcb639a6dfb422

## SetAppProperties

Type
: Function

Sender
: HMI

Purpose
: Enable/disable an application and set authentication data

!!! must

1. Send `SetAppProperties`(enabled=true) if webEngine app is installed.

!!!

!!! may
* Request `UpdateSDL` after setting app properties to enforce SDL to perform PTU.
* Send `SetAppProperties`(enabled=false) to add not-installed webEngine apps to SDL

!!!

!!! info

* The OEM store uses app data in SDL Server (synchronized from SHAID) for SetAppProperties.
* If SDL does not know the `appID` it should trigger a PTU.

!!!

> !!!

Author: tio remove

Author: done in https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/187/commits/33912b35cfa6c0e7fe5044685b2a1b9021816f9f

## SetAppProperties

Type
: Function

Sender
: HMI

Purpose
: Enable/disable an application and set authentication data

!!! must

1. Send `SetAppProperties` (enabled=true) to install a web engine app.

!!!

!!! may
* Send an [UpdateSDL](../../sdl/updatesdl) RPC after setting app properties to force SDL to perform a PTU.
* Send `SetAppProperties` (enabled=false) to add web engine apps to SDL that won't be installed on the HMI

!!!

!!! info

* The OEM store uses app data in SDL Server (synchronized from SHAID) for SetAppProperties.
* If SDL does not know the `appID` it should trigger a PTU.

!!!

### Request

#### Parameters
|Name|Type|Mandatory|Additional|
|:---|:---|:--------|:---------|
|properties|[Common.AppProperties](../../common/structs/#appproperties)|true||

### Response

#### Parameters

This RPC has no additional parameter requirements.

### Example Request

> ```json

Reviewer: This JSON is still very misaligned. Closing brackets should be at the same indentation of the opening tag.
Example:
```json
{
    "opening_tag": {
        "child": "value"
    } <= closing bracket
}
```

I

{
   "id":47,
   "jsonrpc":"2.0",
   "method" : "BasicCommunication.SetAppProperties",
   "params" : {
      "properties":[
         {
            "nicknames":[
               "Hello Sdl"
            
],
            "policyAppID":123456,
            "enabled":true,
            "transportType":"WS",
            "hybridAppPreference":"CLOUD"
         
}
      
],
      "code":0,
      "method":"BasicCommunication.SetAppProperties"
   
}
}
```

### Example Response

```json
{
  "id" : 47,
  "jsonrpc" : "2.0",
  "result" : 
  {
    "code" : 0,
    "method" : "BasicCommunication.SetAppProperties"

>     }

Reviewer: This bracket needs to have two spaces in front of it removed.
